engine/dlib/src/dlib/log.cpp

```diff
@@ -552,6 +556,8 @@ void LogInternal(LogSeverity severity, const char* domain, const char* format, .
 
     va_end(lst);
 
+    DM_SPINLOCK_SCOPED_LOCK(dmLog::g_LogSpinLock);
```

Comment:
The assumption is still that the formatting can be done on the current thread.
It's what happens after this point that needs protecting:
* Custom log hook
* Custom log listener
* Sending log to Profiler

The scope is valid until the end of the function.